core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -442,9 +443,12 @@ private[spark] class MesosClusterScheduler(
     options
   }
 
-  private class ResourceOffer(val offer: Offer, var cpu: Double, var mem: Double) {
+  private class ResourceOffer(
+      val offerId: OfferID,
```

Comment:
in the future I will keep minor refactors like this separate from this patch. It's distracting to the reviewer and is not needed to fix SPARK-10749.